src/runtime/particle-execution-host.ts

```diff
@@ -138,7 +138,7 @@ export class ParticleExecutionHost {
         //
         // Without an auditor on the runtime side that inspects what is being fetched from
         // this store, particles with a reference can access any data of that reference's type.
-        this.GetBackingStoreCallback(store, callback, type.collectionOf(), type.toString(), store.id, storageKey);
+        this.GetBackingStoreCallback(store, callback, type.collectionOf(), type.toString(), String(Math.random())/*store.id*/, storageKey);
```

Comment:
Is this ok to merge?

Reply:
Good catch, I forgot to follow up on that. I remember discussing it with Myk, but I can't find any reference to it now. :(

Will have to page in the details, make an issue ticket, and put a link here.

Reply:
Ok, so I *had* captured the basic problem in #2936.

I tweaked the workaround as Myk suggested and added a note referencing the above ticket.